plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -464,15 +464,16 @@ public Optional<JoinApplicationResult<ConnectorTableHandle>> applyJoin(
         int nextSyntheticColumnId = max(leftHandle.getNextSyntheticColumnId(), rightHandle.getNextSyntheticColumnId());
 
         ImmutableMap.Builder<JdbcColumnHandle, JdbcColumnHandle> newLeftColumnsBuilder = ImmutableMap.builder();
+        OptionalInt maxColumnNameLength = jdbcClient.getMaxColumnNameLength(session);
```

Comment:
I would store `maxColumnNameLength` as field as `Supplier<Integer>`, I am kind of afraid that some JDBC could introduce some penalty here. Even JDBC connection population can case some latency cost, doing this per each column could amplify this.

Also I don't think you need `OptionalInt`. `int` is just enough, no limit can be represented as `Integer.MAX_VALUE`

Reply:
I agree. That's why I asked this question in the beginning at https://github.com/trinodb/trino/pull/18984#discussion_r1322421663.

The only thing then left to verify is that if the Supplier should be a field on the JdbcClient or JdbcMetadata or a singleton bound via Guice? I don't remember if the JdbcClient lives across queries or is scoped to a single query.

Reply:
cached on the BaseJdbcClient now. No need for a supplier anymore.